src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringFromArray = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
```

Comment:
if u do this check u can cover for loop in this and return string once


if (this.length  !== 0) {
      for loop here...
    }
    
    return stringFromArray;

Reply:
agree. I'll change my code. Thanks.